redis/src/cluster.rs

```diff
@@ -791,16 +757,46 @@ fn get_slots(connection: &mut Connection, tls_mode: Option<TlsMode>) -> RedisRes
     Ok(result)
 }
 
-fn build_connection_string(host: &str, port: Option<u16>, tls_mode: Option<TlsMode>) -> String {
-    let host_port = match port {
-        Some(port) => format!("{}:{}", host, port),
-        None => host.to_string(),
+// The node string passed to this function will always be in the format host:port as it is either:
+// - Created by calling ConnectionAddr::to_string (unix connections are not supported in cluster mode)
+// - Returned from redis via the ASK/MOVED response
+fn get_connection_info(node: &str, cluster_params: ClusterParams) -> RedisResult<ConnectionInfo> {
```

Comment:
Can we reduce the repetition here a bit? Lots of potential ways to slice it, but one possibility
```
      let host = split
          .next()
          .ok_or_else(|| (ErrorKind::InvalidClientConfig, "Invalid node string"))?;
      let port = split
          .next()
          .and_then(|val| u16::from_str(val).ok())
          .ok_or_else(|| (ErrorKind::InvalidClientConfig, "Invalid node string"))?;
```

Reply:
I refactored `get_connection_info` method to satisfy your comment with your suggestion and then add my little improvement.